deps/spidershim/test/value.cc

```diff
@@ -778,6 +778,24 @@ TEST(SpiderShim, Date) {
   EXPECT_EQ(time, Date::Cast(*date.ToLocalChecked()->ToObject())->ValueOf());
 }
 
+namespace {
+
+class TestExternalStringResource : public String::ExternalStringResource {
+  public:
+    TestExternalStringResource(const uint16_t* source, size_t length)
+        : data_(source), length_(length) {}
+    ~TestExternalStringResource() {
+      // XXX Check that we get called when the string is finalized.
```

Comment:
> Perhaps instead I can define a TEST(SpiderShim, ExternalStringDestructor) function that checks the result of the destructor. The test harness appears to run the tests in deterministic order (and reconstruct the engine, or at least the isolate, for each test function), so that should work.

Indeed, it does seem to work, per 7f07aef. Is it a reasonable approach?